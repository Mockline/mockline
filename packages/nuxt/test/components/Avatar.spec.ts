import { describe, it, expect } from 'vitest'
import ComponentRender from '../component-render'
import type { TypeOf } from 'zod'
// @ts-ignore
import { MAvatar } from '#components'

describe('Avatar', () => {
  it.each([
    ['basic case with src', {
      props: {
        src: "https://avatars.githubusercontent.com/u/72015679?v=4",
        alt: "Avatar"
      }
    }],
    ['with fallback initials', {
      props: {
        src: "",
        alt: "Johann Cavallucci"
      }
    }],
    ['with icon fallback', {
      props: {
        src: "",
        alt: "Johann Cavallucci",
        icon: "lucide:image"
      }
    }],
    ['with custom text', {
      props: {
        text: "moi"
      }
    }],
    ['with size xs', {
      props: {
        src: "https://avatars.githubusercontent.com/u/72015679?v=4",
        alt: "Avatar",
        size: "xs"
      }
    }],
    ['with size sm', {
      props: {
        src: "https://avatars.githubusercontent.com/u/72015679?v=4",
        alt: "Avatar",
        size: "sm"
      }
    }],
    ['with size md', {
      props: {
        src: "https://avatars.githubusercontent.com/u/72015679?v=4",
        alt: "Avatar",
        size: "md"
      }
    }],
    ['with size lg', {
      props: {
        src: "https://avatars.githubusercontent.com/u/72015679?v=4",
        alt: "Avatar",
        size: "lg"
      }
    }],
    ['with size xl', {
      props: {
        src: "https://avatars.githubusercontent.com/u/72015679?v=4",
        alt: "Avatar",
        size: "xl",
      }
    }],
    ['with no props', { }],
  ])('renders %s correctly', async (nameOrHtml: string, options: TypeOf<typeof MAvatar.props>) => {
    const html = await ComponentRender(nameOrHtml, options, MAvatar)
    expect(html).toMatchSnapshot()
  })
})
