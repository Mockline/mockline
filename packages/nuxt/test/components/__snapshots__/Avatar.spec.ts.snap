// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Avatar > renders basic case with src correctly 1`] = `"<span class="relative inline-flex items-center justify-center rounded-full size-10"><img role="img" src="https://avatars.githubusercontent.com/u/72015679?v=4" is="img" class="size-full rounded-[inherit] object-cover ring-[2px] ring-[var(--content-area-bg)]" style="display: none;"><!----></span>"`;

exports[`Avatar > renders with custom text correctly 1`] = `"<span class="relative inline-flex items-center justify-center rounded-full size-10"><!--v-if--><!----></span>"`;

exports[`Avatar > renders with fallback initials correctly 1`] = `"<span class="relative inline-flex items-center justify-center rounded-full size-10"><!--v-if--><!----></span>"`;

exports[`Avatar > renders with icon fallback correctly 1`] = `"<span class="relative inline-flex items-center justify-center rounded-full size-10"><!--v-if--><!----></span>"`;

exports[`Avatar > renders with no props correctly 1`] = `"<span class="relative inline-flex items-center justify-center rounded-full size-10"><!--v-if--><!----></span>"`;

exports[`Avatar > renders with size lg correctly 1`] = `"<span class="relative inline-flex items-center justify-center rounded-full size-12"><img role="img" src="https://avatars.githubusercontent.com/u/72015679?v=4" is="img" class="size-full rounded-[inherit] object-cover ring-[2px] ring-[var(--content-area-bg)]" style="display: none;"><!----></span>"`;

exports[`Avatar > renders with size md correctly 1`] = `"<span class="relative inline-flex items-center justify-center rounded-full size-10"><img role="img" src="https://avatars.githubusercontent.com/u/72015679?v=4" is="img" class="size-full rounded-[inherit] object-cover ring-[2px] ring-[var(--content-area-bg)]" style="display: none;"><!----></span>"`;

exports[`Avatar > renders with size sm correctly 1`] = `"<span class="relative inline-flex items-center justify-center rounded-full size-8"><img role="img" src="https://avatars.githubusercontent.com/u/72015679?v=4" is="img" class="size-full rounded-[inherit] object-cover ring-[2px] ring-[var(--content-area-bg)]" style="display: none;"><!----></span>"`;

exports[`Avatar > renders with size xl correctly 1`] = `"<span class="relative inline-flex items-center justify-center rounded-full size-14"><img role="img" src="https://avatars.githubusercontent.com/u/72015679?v=4" is="img" class="size-full rounded-[inherit] object-cover ring-[2px] ring-[var(--content-area-bg)]" style="display: none;"><!----></span>"`;

exports[`Avatar > renders with size xs correctly 1`] = `"<span class="relative inline-flex items-center justify-center rounded-full size-6"><img role="img" src="https://avatars.githubusercontent.com/u/72015679?v=4" is="img" class="size-full rounded-[inherit] object-cover ring-[2px] ring-[var(--content-area-bg)]" style="display: none;"><!----></span>"`;
